pkg/kv/kvserver/client_raft_test.go

```diff
@@ -5572,6 +5572,24 @@ func TestProcessSplitAfterRightHandSideHasBeenRemoved(t *testing.T) {
 			_, store0Exists := rhsInfo.Desc.GetReplicaDescriptor(1)
 			require.True(t, store0Exists)
 
+			// Ensure that store 0 has created the RHS replica in memory from raft
+			// messages before we proceed. This is necessary for the tombstone to be
+			// written when the learner snapshot is rejected. Without this, there's a
+			// race where the learner snapshot might arrive before any raft heartbeats,
+			// causing the test to flake.
+
+			// Ensure that we have an (uninitialized) replica on Store 0 before
+			// proceeding. There should be one soon in response to a leader election
+			// for the split off RHS. Without this, we'll not write a tombstone as
+			// part of rejecting the learner snapshot below.
```

Comment:
The 2 paragraphs seem to be talking about same things repetitively. Squash into one?

```

```